lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2879,7 +2879,16 @@ Function *SPIRVToLLVM::transFunction(SPIRVFunction *BF) {
     BA->foreachAttr([&](SPIRVFuncParamAttrKind Kind) {
       if (Kind == FunctionParameterAttributeNoWrite)
         return;
-      F->addAttribute(I->getArgNo() + 1, SPIRSPIRVFuncParamAttrMap::rmap(Kind));
+      Attribute::AttrKind LLVMKind = SPIRSPIRVFuncParamAttrMap::rmap(Kind);
```

Comment:
Am I right that this change is covered by some existing tests? I.e. we do not have an exact check that we generated typed attribute, but since this is a requirement of LLVM and we have tests that use such attributes we have this change covered by the fact that some existing tests are not crashing during backward translation, right?

Reply:
Yes, I believe that's a very clean description.

Before the change was up, LLVM validation was failing when assembling the `.bc` file due to attributes' lacking the now-mandatory type; other tests were crashing on a `SPIRVReader`-stemming assertion because the LLVM library had been expecting a non-`nullptr` type.

The fact that the `isa<> applied to nullptr` assertion no longer fires can prove that the type gets set; the fact that `llvm-dis` is able to produce a textual IR file proves that `.bc` passes the validation.